R/cobertura.R

```diff
@@ -0,0 +1,181 @@
+#' Create a Cobertura XML file
+#' @param cov the coverage object returned from \code{\link{package_coverage}}
+#' @param filename the name of the Cobertura XML file
+#' @author Willem Ligtenberg
+#' @export
+to_cobertura <- function(cov, filename = 'cobertura.xml'){
+  xmlDocument = XML::newXMLDoc()
+  
+  df <- tally_coverage(cov, by = "line")
+  
+  top <- XML::newXMLNode(name = "coverage", 
+      attrs = c(
+          "line-rate" = as.character(percent_coverage(df, by = "line")/100),
+          "branch-rate" = "0",
+          version = sessionInfo()$otherPkgs$covr$Version,
+          timestamp = as.character(Sys.time())),
+      parent = xmlDocument)
+  
+  # Add sources
+  sources <- XML::newXMLNode(
+      name = "sources", 
+      parent = top)
+  sourceFiles <- unique(covr:::display_name(cov))
+  for(f in sourceFiles){
+    source <- XML::newXMLNode(
+        name = "source", 
+        text = f,
+        parent = sources)
+  }
+  
+  # Add packages
+  packages <- XML::newXMLNode(
+      name = "packages", 
+      parent = top)
+  
+  packageName <- sub(".*?/(.+?)/R/.+?\\.[Rr]", "\\1", get("filename", attr(cov[[1]]$srcref, 'srcfile')))
```

Comment:
The coverage objects have a package object attached, so you can retrieve the package name with

```r
attr(x, "package")$package
```